event-manager/src/main/java/evrentan/community/eventmanager/impl/EventServiceImpl.java

```
@@ -116,13 +114,22 @@ public Event getEvent(UUID id) {
      * @since 1.0.0
      */
     @Override
-    public Event updateEvent(UUID id, Event event) {
-        if(!Objects.equals(id, event.getId()))
-            throw new IllegalArgumentException("Ids do not match");
-        if(!this.eventRepository.existsById(id))
-            throw new NoSuchElementException("Event not found");
+    public ResponseEntity<Event> updateEvent(UUID id, Event event) {
+      if (Objects.isNull(id) || Objects.isNull(event) || Objects.equals(id, event.getId()))
+          return ResponseEntity.badRequest().build();
+
+        Optional<EventEntity> existingEvent = eventRepository.findById(id);
+
+        if (existingEvent.isEmpty())
+            return ResponseEntity.notFound().build();
```

instead of returning Response entity, throw a custom exception here.